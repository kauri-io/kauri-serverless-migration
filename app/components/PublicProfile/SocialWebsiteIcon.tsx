import * as React from "react";
import styled from "../../lib/styled-components";

const InvertSocialIcon = styled.img`
  filter: invert(100%);
`;

interface ISocialWebsiteIconProps {
  brand: string;
  height?: number;
  socialURL?: string;
}

const SocialWebsiteIcon: React.SFC<ISocialWebsiteIconProps> = ({
  height = 20,
  brand,
  socialURL,
}) => (
  <a target="_blank" href={socialURL}>
    <InvertSocialIcon
      height={height}
      width={height}
      src={`https://unpkg.com/simple-icons@latest/icons/${brand}.svg`}
    />
  </a>
);
export default SocialWebsiteIcon;
