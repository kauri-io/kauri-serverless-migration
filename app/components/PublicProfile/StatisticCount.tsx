import styled from 'styled-components'
import { cond } from 'ramda'

interface IPageTypeProps {
    pageType: string
}

const Count = styled.h3<IPageTypeProps>`
    color: white;
    font-size: ${props => props.theme.fontSizes[7]}px;
    font-weight: ${props => props.theme.fontWeight[2]};
    margin: 0px;
    margin-bottom: ${props => props.theme.space[0]}px;
    opacity: ${props =>
        typeof props.pageType === 'string' &&
        props.pageType !== 'CollectionPage'
            ? 0.2
            : 1.0};
`

const SectionContainer = styled.section`
    display: flex;
    flex-direction: column;
    align-items: center;
    max-width: 150px;
`

const Name = styled.span<IPageTypeProps>`
    color: white;
    font-size: ${props => props.theme.fontSizes[0]}px;
    font-weight: ${props => props.theme.fontWeight[3]};
    opacity: ${props =>
        typeof props.pageType === 'string' &&
        props.pageType !== 'CollectionPage'
            ? 0.2
            : 1.0};
    text-transform: uppercase;
`

const pluraliseCount = (payload: number) =>
    cond([
        [
            count => count > 1000000,
            count => `${(count / 1000000).toPrecision(2)}M`,
        ],
        [
            count => count > 1000 && count < 1000000,
            count => `${(count / 1000).toPrecision(2)}K`,
        ],
        [count => count < 1000, count => String(count)],
    ])(payload)

const lastStringValue = (name: string) =>
    name && name.length > 0 && name[name.length - 1]

const pluraliseName = (payload: { count: number; name: string }) =>
    cond([
        [
            ({ count, name }) => count > 1 && lastStringValue(name) === 's',
            ({ name }) => name,
        ],
        [
            ({ count, name }) => count > 1 && lastStringValue(name) !== 's',
            ({ name }) => `${name}s`,
        ],
        [
            ({ count, name }) => count === 1 && lastStringValue(name) === 's',
            ({ name }) => name.substring(0, name.length - 1),
        ],
        [({ name }) => typeof name === 'string', ({ name }) => name],
    ])(payload)

interface IProps {
    name: string
    count: number
    pageType: string
}

const Container: React.SFC<IProps> = ({ name, count, pageType }) => (
    <SectionContainer key={name}>
        <Count pageType={pageType}>{pluraliseCount(count)}</Count>
        <Name pageType={pageType}>{pluraliseName({ name, count })}</Name>
    </SectionContainer>
)

export default Container
