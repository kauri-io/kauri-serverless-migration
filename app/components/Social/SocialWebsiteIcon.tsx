import styled from 'styled-components'

const InvertSocialIcon = styled.img`
    filter: invert(100%);
`

interface ISocialWebsiteIconProps {
    brand: string
    height?: number
    socialURL?: string
    invert?: boolean
    className?: string
}

const SocialWebsiteIcon: React.SFC<ISocialWebsiteIconProps> = ({
    height = 20,
    brand,
    socialURL,
    invert = true,
    className,
}) => {
    const getIconUrl = brand => {
        return `https://unpkg.com/simple-icons@latest/icons/${brand}.svg`
    }
    return (
        <>
            <a target="_blank" href={socialURL}>
                {invert && (
                    <InvertSocialIcon
                        alt={brand}
                        height={height}
                        width={height}
                        src={getIconUrl(brand)}
                        className={className}
                    />
                )}

                {!invert && (
                    <img
                        alt={brand}
                        height={height}
                        width={height}
                        src={getIconUrl(brand)}
                        className={className}
                    />
                )}
            </a>
        </>
    )
}
export default SocialWebsiteIcon
