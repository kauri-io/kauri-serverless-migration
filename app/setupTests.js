import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
