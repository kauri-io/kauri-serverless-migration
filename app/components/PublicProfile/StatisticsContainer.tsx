import * as React from "react";
import styled from "styled-components";
import StatisticCount from "./StatisticCount";

const StatisticsContainer = styled.section`
  display: flex;
  flex-direction: row;
  > :not(:first-child) {
    margin-left: ${props => props.theme.space[4]}px;
  }
`;

interface IStatistic {
  count: number;
  name: string;
}

interface IProps {
  pageType?: "CreateCollectionPage" | "CreateCommunityPage" | "CollectionPage";
  statistics: IStatistic[];
}

const Container: React.SFC<IProps> = props => {
  const { statistics, pageType } = props;
  return (
    <StatisticsContainer>
      {statistics.map(({ name, count }) => (
        <StatisticCount
          pageType={pageType}
          key={name}
          name={name}
          count={count}
        />
      ))}
    </StatisticsContainer>
  );
};

export default Container;
