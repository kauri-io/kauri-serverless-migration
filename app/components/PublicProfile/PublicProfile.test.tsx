import SocialWebsiteIcon from './SocialWebsiteIcon'
import StatisticCount from './StatisticCount'
import StatisticsContainer from './StatisticsContainer'
import { mountWithRedux } from '../../setupTests'

describe('components/PublicProfile/SocialWebsiteIcon', () => {
    it('should match snapshot', () => {
        const SocialWebsiteIconProps = {
            brand: 'twitter',
        }

        const wrapper = mountWithRedux(
            <SocialWebsiteIcon {...SocialWebsiteIconProps} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})

describe('components/PublicProfile/StatisticCount', () => {
    it('should match snapshot', () => {
        const StatisticCountProps = {
            name: 'test',
            count: 4,
            pageType: 'collection',
        }

        const wrapper = mountWithRedux(
            <StatisticCount {...StatisticCountProps} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})

describe('components/PublicProfile/StatisticsContainer', () => {
    it('should match snapshot', () => {
        const StatisticsContainerProps = {
            statistics: [
                {
                    name: 'test',
                    count: 4,
                    pageType: 'collection',
                },
                {
                    name: 'test2',
                    count: 4,
                    pageType: 'collection',
                },
            ],
        }

        const wrapper = mountWithRedux(
            <StatisticsContainer {...StatisticsContainerProps} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})
