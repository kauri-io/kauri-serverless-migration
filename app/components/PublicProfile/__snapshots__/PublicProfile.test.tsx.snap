// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PublicProfile/SocialWebsiteIcon should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <SocialWebsiteIcon
      brand="twitter"
    >
      <a
        target="_blank"
      >
        <SocialWebsiteIcon__InvertSocialIcon
          height={20}
          src="https://unpkg.com/simple-icons@latest/icons/twitter.svg"
          width={20}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SocialWebsiteIcon__InvertSocialIcon-sc-1n3fjnh-0",
                  "isStatic": true,
                  "lastClassName": "dyAGBL",
                  "rules": Array [
                    "filter:invert(100%);",
                  ],
                },
                "displayName": "SocialWebsiteIcon__InvertSocialIcon",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SocialWebsiteIcon__InvertSocialIcon-sc-1n3fjnh-0",
                "target": "img",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height={20}
            src="https://unpkg.com/simple-icons@latest/icons/twitter.svg"
            width={20}
          >
            <img
              className="SocialWebsiteIcon__InvertSocialIcon-sc-1n3fjnh-0 dyAGBL"
              height={20}
              src="https://unpkg.com/simple-icons@latest/icons/twitter.svg"
              width={20}
            />
          </StyledComponent>
        </SocialWebsiteIcon__InvertSocialIcon>
      </a>
    </SocialWebsiteIcon>
  </ThemeProvider>
</Provider>
`;

exports[`components/PublicProfile/StatisticCount should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <Container
      count={4}
      name="test"
      pageType="collection"
    >
      <StatisticCount__SectionContainer
        key="test"
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "StatisticCount__SectionContainer-sc-1xyux9h-1",
                "isStatic": true,
                "lastClassName": "hBrKEM",
                "rules": Array [
                  "display:flex;flex-direction:column;align-items:center;max-width:150px;",
                ],
              },
              "displayName": "StatisticCount__SectionContainer",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "StatisticCount__SectionContainer-sc-1xyux9h-1",
              "target": "section",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <section
            className="StatisticCount__SectionContainer-sc-1xyux9h-1 hBrKEM"
          >
            <StatisticCount__Count
              pageType="collection"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "StatisticCount__Count-sc-1xyux9h-0",
                      "isStatic": false,
                      "lastClassName": "dTPpIr",
                      "rules": Array [
                        "color:white;font-size:",
                        [Function],
                        "px;font-weight:",
                        [Function],
                        ";margin:0px;margin-bottom:",
                        [Function],
                        "px;opacity:",
                        [Function],
                        ";",
                      ],
                    },
                    "displayName": "StatisticCount__Count",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "StatisticCount__Count-sc-1xyux9h-0",
                    "target": "h3",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                pageType="collection"
              >
                <h3
                  className="StatisticCount__Count-sc-1xyux9h-0 dTPpIr"
                >
                  4
                </h3>
              </StyledComponent>
            </StatisticCount__Count>
            <StatisticCount__Name
              pageType="collection"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "StatisticCount__Name-sc-1xyux9h-2",
                      "isStatic": false,
                      "lastClassName": "ivYcUc",
                      "rules": Array [
                        "color:white;font-size:",
                        [Function],
                        "px;font-weight:",
                        [Function],
                        ";opacity:",
                        [Function],
                        ";text-transform:uppercase;",
                      ],
                    },
                    "displayName": "StatisticCount__Name",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "StatisticCount__Name-sc-1xyux9h-2",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                pageType="collection"
              >
                <span
                  className="StatisticCount__Name-sc-1xyux9h-2 ivYcUc"
                >
                  tests
                </span>
              </StyledComponent>
            </StatisticCount__Name>
          </section>
        </StyledComponent>
      </StatisticCount__SectionContainer>
    </Container>
  </ThemeProvider>
</Provider>
`;

exports[`components/PublicProfile/StatisticsContainer should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <Container
      statistics={
        Array [
          Object {
            "count": 4,
            "name": "test",
            "pageType": "collection",
          },
          Object {
            "count": 4,
            "name": "test2",
            "pageType": "collection",
          },
        ]
      }
    >
      <StatisticsContainer>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "StatisticsContainer-sc-18atadq-0",
                "isStatic": false,
                "lastClassName": "kulLJ",
                "rules": Array [
                  "display:flex;flex-direction:row;>:not(:first-child){margin-left:",
                  [Function],
                  "px;}",
                ],
              },
              "displayName": "StatisticsContainer",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "StatisticsContainer-sc-18atadq-0",
              "target": "section",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <section
            className="StatisticsContainer-sc-18atadq-0 kulLJ"
          >
            <Container
              count={4}
              key="test"
              name="test"
              pageType="undefined"
            >
              <StatisticCount__SectionContainer
                key="test"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "StatisticCount__SectionContainer-sc-1xyux9h-1",
                        "isStatic": true,
                        "lastClassName": "hBrKEM",
                        "rules": Array [
                          "display:flex;flex-direction:column;align-items:center;max-width:150px;",
                        ],
                      },
                      "displayName": "StatisticCount__SectionContainer",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "StatisticCount__SectionContainer-sc-1xyux9h-1",
                      "target": "section",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <section
                    className="StatisticCount__SectionContainer-sc-1xyux9h-1 hBrKEM"
                  >
                    <StatisticCount__Count
                      pageType="undefined"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "StatisticCount__Count-sc-1xyux9h-0",
                              "isStatic": false,
                              "lastClassName": "dTPpIr",
                              "rules": Array [
                                "color:white;font-size:",
                                [Function],
                                "px;font-weight:",
                                [Function],
                                ";margin:0px;margin-bottom:",
                                [Function],
                                "px;opacity:",
                                [Function],
                                ";",
                              ],
                            },
                            "displayName": "StatisticCount__Count",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "StatisticCount__Count-sc-1xyux9h-0",
                            "target": "h3",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        pageType="undefined"
                      >
                        <h3
                          className="StatisticCount__Count-sc-1xyux9h-0 dTPpIr"
                        >
                          4
                        </h3>
                      </StyledComponent>
                    </StatisticCount__Count>
                    <StatisticCount__Name
                      pageType="undefined"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "StatisticCount__Name-sc-1xyux9h-2",
                              "isStatic": false,
                              "lastClassName": "ivYcUc",
                              "rules": Array [
                                "color:white;font-size:",
                                [Function],
                                "px;font-weight:",
                                [Function],
                                ";opacity:",
                                [Function],
                                ";text-transform:uppercase;",
                              ],
                            },
                            "displayName": "StatisticCount__Name",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "StatisticCount__Name-sc-1xyux9h-2",
                            "target": "span",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        pageType="undefined"
                      >
                        <span
                          className="StatisticCount__Name-sc-1xyux9h-2 ivYcUc"
                        >
                          tests
                        </span>
                      </StyledComponent>
                    </StatisticCount__Name>
                  </section>
                </StyledComponent>
              </StatisticCount__SectionContainer>
            </Container>
            <Container
              count={4}
              key="test2"
              name="test2"
              pageType="undefined"
            >
              <StatisticCount__SectionContainer
                key="test2"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "StatisticCount__SectionContainer-sc-1xyux9h-1",
                        "isStatic": true,
                        "lastClassName": "hBrKEM",
                        "rules": Array [
                          "display:flex;flex-direction:column;align-items:center;max-width:150px;",
                        ],
                      },
                      "displayName": "StatisticCount__SectionContainer",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "StatisticCount__SectionContainer-sc-1xyux9h-1",
                      "target": "section",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <section
                    className="StatisticCount__SectionContainer-sc-1xyux9h-1 hBrKEM"
                  >
                    <StatisticCount__Count
                      pageType="undefined"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "StatisticCount__Count-sc-1xyux9h-0",
                              "isStatic": false,
                              "lastClassName": "dTPpIr",
                              "rules": Array [
                                "color:white;font-size:",
                                [Function],
                                "px;font-weight:",
                                [Function],
                                ";margin:0px;margin-bottom:",
                                [Function],
                                "px;opacity:",
                                [Function],
                                ";",
                              ],
                            },
                            "displayName": "StatisticCount__Count",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "StatisticCount__Count-sc-1xyux9h-0",
                            "target": "h3",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        pageType="undefined"
                      >
                        <h3
                          className="StatisticCount__Count-sc-1xyux9h-0 dTPpIr"
                        >
                          4
                        </h3>
                      </StyledComponent>
                    </StatisticCount__Count>
                    <StatisticCount__Name
                      pageType="undefined"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "StatisticCount__Name-sc-1xyux9h-2",
                              "isStatic": false,
                              "lastClassName": "ivYcUc",
                              "rules": Array [
                                "color:white;font-size:",
                                [Function],
                                "px;font-weight:",
                                [Function],
                                ";opacity:",
                                [Function],
                                ";text-transform:uppercase;",
                              ],
                            },
                            "displayName": "StatisticCount__Name",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "StatisticCount__Name-sc-1xyux9h-2",
                            "target": "span",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        pageType="undefined"
                      >
                        <span
                          className="StatisticCount__Name-sc-1xyux9h-2 ivYcUc"
                        >
                          test2s
                        </span>
                      </StyledComponent>
                    </StatisticCount__Name>
                  </section>
                </StyledComponent>
              </StatisticCount__SectionContainer>
            </Container>
          </section>
        </StyledComponent>
      </StatisticsContainer>
    </Container>
  </ThemeProvider>
</Provider>
`;
